src/TrackGenerator.cpp

```diff
@@ -559,7 +605,9 @@ void TrackGenerator::generateTracks() {
     }
   }
 
+  /* Initialize the track boundary conditions and set the track UIDs */
   initializeBoundaryConditions();
+  initializeTrackPeriodicCycleIndices();
```

Comment:
I second @wbinventor's suggestion of a single with a parameter as long as it wouldn't be too tough